### PR TITLE
METRO-61: MODS clean up self-transform to remove unused elements.

### PR DESCRIPTION
**Pull request by matthewperry**
Impemented hook_islandora_derivative to apply a clean up self transform to the mods datastream.

### CODE REVIEW

**Comment by matthewperry** [comment]
Using the XSL from the xml_forms module instead of duplicating it to prevent having to maintain two of copies of the same file.

File: includes/clean_mods_datastream.inc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+function metrodora_clean_mods_datastream(AbstractObject $object, $force, $hook) {
+  if (!isset($object['MODS'])) {
+    return;
+  }
+
+  $original_doc = new DOMDocument();
+  $original_doc->loadXML($object['MODS']->content);
+  $transformed_doc = metrodora_run_xslt($original_doc,
+    drupal_get_path('module', 'xml_forms') .
+    '/builder/self_transforms/islandora_cleanup_mods_extended.xsl');
```
Comment (adam-vessey) [inline includes/clean_mods_datastream.inc]: Probably better to target the `xml_form_builder` module, instead of hitting relative to `xml_forms`.

Comment (matthewperry) [inline includes/clean_mods_datastream.inc]: Modified target to use xml_form_builder as module.

File: metrodora.module [hunk 2]
```diff
@@ -493,3 +493,23 @@ function metrodora_islandora_compound_object_management_control_paths() {
...
+function metrodora_islandora_derivative(AbstractObject $object = NULL) {
+
+  $derivatives[] = array(
+    'source_dsid' => 'MODS',
+    'destination_dsid' => 'MODS',
+    'function' => array(
+      'metrodora_clean_mods_datastream',
+    ),
+    'file' => drupal_get_path('module', 'metrodora') . '/includes/clean_mods_datastream.inc',
+  );
```
Comment (adam-vessey) [inline metrodora.module:514]: Should probably be weighted above the [default DC stuff](https://github.com/Islandora/islandora_xml_forms/blob/7.x/builder/xml_form_builder.module#L738-L743) (default weight is assumed to be 0, go to -1)? Also, could we get `$derivatives` explicitly defined as an empty array at the top of the function? :P

**Pull request merged**